### ISSUE TITLE
Tab completion not working when loaded through antigen

### ISSUE THREAD

**Issue by petelewis** [zsh-users/antigen#280]
Hi,

I am new to antigen and just playing around with wrapping up some functions in antigen bundles. I have found that tab-complete doesn't work for me when I load a simple function through antigen though.

Here's an excerpt from my `.zshrc` file that saves me a few keystrokes:

    x() {
      ~/.screenlayout/$1
    }
    compdef '_path_files -W ~/.screenlayout/' x`

This works fine, and `x<space><tab>` lists all the files inside `~/.screenlayout/`.

I then put this same excerpt into a file called `x.zsh` inside `~/src/zsh-x/` and loaded it (in a new shell where this excerpt was not present in the `.zshrc`) with `antigen bundle ~/src/zsh-x`

When I do this, the `x` function still works fine, but tab completion just lists all the files in the current directory (i.e. doesn't work). I tried this with and without `antigen apply`, and it made no difference.

What am I doing wrong?

I'm on zsh 5.2 on Archlinux, with a self-updated antigen as of a few minutes ago.

Thanks. Apologies if this isn't the right place to ask this.

**Comment by desyncr** [comment]
Doing it interactively it seems to work:

    antigen bundle /path/to/zsh-x x.zsh

But it seems that adding it to antigen config (.zshrc, or .antigenrc) it seems to break it.

I'm going to take a deeper look at it. Thanks for reporting it.

**Comment by desyncr** [comment]
Well, now it looks like it's a zcompdump issue again.

Do try removing it:

    mv ~/.zcompdump ~/.zcompdump.1

This issue should have been resolved at https://github.com/zsh-users/antigen/pull/273 but it seems it still present, so I'll leave this issue open.

Thanks again for reporting.

**Comment by petelewis** [comment]
Thanks for the quick response.

Yep, I tried deleting .compdump (I don't have a .zcompdump.1), and it made no difference.

But, different from you, loading it interactively doesn't work for me either.

I am a little bit unclear on the syntax of the bundle command though. I was under the impression that the preferred thing to do is to call the file `x.plugin.zsh` (although perhaps that doesn't matter). Assuming I call the file `x.zsh`, as you have done, trying to load it with `antigen bundle /path/to/zsh-x x.zsh` doesn't even load the function, whereas `antigen bundle /path/to/zsh-x` does. I'm not sure if this is related.

In any case, I can't get tab completion to load with antigen interactively or otherwise.

Thanks for looking into it.

**Comment by desyncr** [comment]
Hello there. Running antigen `1.2.0` and everything seems alright from my side.

Copied and pasted:

    x() {
      ~/Projects/$1
    }
    compdef '_path_files -W ~/Projects/' x


(changed path, but the snippet should be the same regardless of it)

Created it under `~/Playground/zsh-x` with `x.zsh` as file name.


Interactively running:

    antigen bundle ~/Playground/zsh-x # (or antigen bundle ~/Playground/zsh-x x.zsh)

Loaded the bundle correctly and tab completion worked as expected.

Adding

    antigen bundle ~/Playground/zsh-x

To `.antigenrc` (`.zshrc` or wherever you configure antigen) also did the trick. BUT in two different machines (Fedora & Mac OS X) I had to manually remove `.zcompdump` file to actually see tab completion.

Regarding, `bundle` command syntax it's as described in https://github.com/zsh-users/antigen#antigen-bundle, namely:

    antigen bundle [<url> [<loc>]]


Basically, using `<loc>`, you're forcing antigen to load whatever thing you're referring to, ie:

         antigen bundle someuser/somerepo somefile.py

Would try to source `somefile.py`. Meanwhile with:

         antigen bundle someuser/somerepo

Antigen would go through this matching:

- *.plugin.zsh
- init.zsh
- *.zsh
- *.sh

In any case .plugin.zsh is NOT necessary or mandatory at all.

**Comment by chaoyangnz** [comment]
why close? it is still happening with zsh 5.8
